V10 Compatibility

MiniGrief opened the pull request:
Haven't checked over everything. Only changed what needed changing to make it appear functional.

MiniGrief committed:
V10 Compatibility

Haven't checked over everything. Only changed what needed changing to make it appear functional.

MiniGrief committed:
Update module.json

Marks module as compatible with V10

MiniGrief committed:
Remove V8/9 Compatibility

Current changes don't allow for V8/9 compatiblity but something similar to "isNewerVersion("0.10.0", game.version)" can be used to add this compatibility back.
I currently can't run any older version to add/test this functionality.

MiniGrief commented:
Not sure of the exact use for it, but around line 115 is using a removed variable for getting the current game version. Didn't want to modify due to not being able to run older versions of Foundry right now so I can't test if it would break anything to 'fix' this. If it needs fixing, it needs changing to "game.version" instead of "game.data.version".

Current changes also don't allow for V8/9 compatiblity but something similar to "isNewerVersion("0.10.0", game.version)" can be used to add this compatibility back.

MiniGrief committed:
Updating more deprecated stuff

mhilbrunner committed:
Prepare version 1.7.0

mhilbrunner committed:
Fix default wildcard image

mhilbrunner committed:
Fix manifest link in README

mhilbrunner commented:
I applied this fix and also fixed the default image feature in [my fork](https://github.com/mhilbrunner/token-hud-wildcard). Seems to work fine, besides some deprecation warnings in console. I published a release as a temporary workaround until this original repository is updated, if someone wants to install it, use this manifest:
`https://raw.githubusercontent.com/mhilbrunner/token-hud-wildcard/master/module.json`
Alternatively, download a .zip from the [release page](https://github.com/mhilbrunner/token-hud-wildcard/releases).
Thanks for doing most of the work!

mhilbrunner committed:
Update README.md

mhilbrunner committed:
Fix README typo

MiniGrief committed:
Further deprecation warning fixes

These were the last few things I could find which were throwing deprecation warnings. Not sure if there's anything else, but they would be easy fixes like these were.

MiniGrief commented:
Some changes made in the fork from @mhilbrunner will have to be changed back to work in the original version but they contained a few extra fixes and bonuses so felt it was worth making them part of this pull request.

MiniGrief committed:
Quick fix for incorrect scaling behaviour introduced in v10

There surely should be a better way than only looking at either x or y but while you can use _source to still access scale, it's deprecated and throws a warning.

cs96and commented on main.js:
```diff
@@ -16,143 +16,139 @@ const tokenHUDWildcard = {
+            if (defaultValue !== '' && parent?.actor?.prototypeToken?.randomImg) {
+                /*const dimensions = getTokenDimensions(parent, defaultValue);
+                let updateInfo = { img: defaultValue, ...dimensions };*/
+                parent.data.update({"texture.src" : defaultValue});
```
This should be changed to...
```javascript
parent.updateSource({"texture.src" : defaultValue});
```
This fixes a warning that appears when you drag an actor with a wildcard token onto the scene.

cs96and commented on main.js:
```diff
@@ -16,143 +16,139 @@ const tokenHUDWildcard = {
-                let updateInfo = { img: defaultValue , ...dimensions }
-                mergeObject(data, updateInfo)
-                parent.data.update(updateInfo)
+            const defaultValue = parent?.actor?.prototypeToken?.flags['token-hud-wildcard'] ? parent.actor.prototypeToken.flags['token-hud-wildcard'].default : '';
+            if (defaultValue !== '' && parent?.actor?.prototypeToken?.randomImg) {
```
This can be changed to...
```javascript
const defaultValue = parent?.actor?.prototypeToken?.getFlag('token-hud-wildcard', 'default');
if (defaultValue && parent?.actor?.prototypeToken?.randomImg) {
```
Just a style thing, but it makes the code much easier to read

MiniGrief committed:
Further fixes from @cs96and

MiniGrief commented:
@cs96and I added your changes to the fork I made. Sorry if it's not exactly the right way to do it as I don't use github too often so didn't know how to merge your changes directly into that fork.

cs96and commented:
I only added those changes as comments (not a proper commit), so I think manually adding them to your fork is the only way to do it.

cs96and commented:
@mhilbrunner, can you update your fork with my latest changes please? (they were added to @MiniGrief's fork)

mhilbrunner commented:
@cs96and Done, thanks! 🎅 https://github.com/mhilbrunner/token-hud-wildcard/releases/tag/1.7.2

mhilbrunner commented:
https://github.com/mhilbrunner/token-hud-wildcard is now updated for v12.
